README.md

```diff
@@ -1,68 +1,107 @@
-(https://github.com/BodenmillerGroup/ImcPluginsCP, release v4.2.1).
 
-This repository showcases the basis of the workflow with step-by-step instructions. To run this more automatized, we recommend
-our Snakemake implementation: https://github.com/BodenmillerGroup/ImcSegmentationSnakemake
+The pipeline is based on [CellProfiler](http://cellprofiler.org/) (tested v4.2.1) for segmentation and [Ilastik](http://ilastik.org/) (tested v1.3.3post3) for pixel classification. 
+It is streamlined by using the `imcsegpipe` python package available via this repository as well as custom CellProfiler modules ([ImcPluginsCP](https://github.com/BodenmillerGroup/ImcPluginsCP), release v4.2.1).
 
-This pipeline was developed in the Bodenmiller laboratory at the University of Zurich (http://www.bodenmillerlab.org/)
-to segment hundreds of highly multiplexed imaging mass cytometry (IMC) images.
-However we think that this concept also works well for other multiplexed imaging modalities..
+This repository showcases the basis of the workflow with step-by-step instructions. 
+As an alternative and dockerized version of the pipeline, check out [steinbock](https://github.com/BodenmillerGroup/steinbock).
 
-The document to start can be found at 'scripts/imc_preprocessing.ipynb'
-(https://nbviewer.jupyter.org/github/BodenmillerGroup/ImcSegmentationPipeline/blob/development/scripts/imc_preprocessing.ipynb).
-The pdf found in 'Documentation/201709_imctools_guide.pdf' is still conceptually valid, however the installation
-procedures described are outdated. Please follow the instructions in the imc_preprocessing.ipynb document!
+This pipeline was developed in the Bodenmiller laboratory at the University of Zurich ([www.bodenmillerlab.com](https://www.bodenmillerlab.com/)) to segment hundreds of highly multiplexed imaging mass cytometry (IMC) images.
+The concepts applied here to IMC data can also be transfered to data generated by other highly multiplexed imaging modalities.
+
+For a general overview on IMC as technology and data processing tasks, please refer to [bodenmillergroup.github.io/IMCWorkflow](https://bodenmillergroup.github.io/IMCWorkflow/).
+
+## Usage
+
+For the main part of the analysis, you will need to install [Ilastik](https://www.ilastik.org/download.html) and [CellProfiler](https://cellprofiler.org/releases).
+
+Before being able to pre-process the data, you will need to setup the environment:
+
+1. [Install conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/)
+
+2. Clone the repository: 
+
+```bash
+git clone https://github.com/BodenmillerGroup/ImcSegmentationPipeline.git
+```
+
+3. Setup the conda environment: 
+
+```bash
+cd ImcSegmentationPipeline
+conda env create -f environment.yml
+```
+
+4. Obtain the IMC-specific [CellProfiler plugins](https://github.com/BodenmillerGroup/ImcPluginsCP) by directly downloading them via:
+
+```bash
+git submodule update --init --recursive
```

Comment:
Why not doing a `git clone --recursive` initially?

Reply:
Also, which version of the plugins is included atm? Should we tag it?

Reply:
Apparently git submodules don't support tags or commits (see [here](https://stackoverflow.com/questions/1777854/how-can-i-specify-a-branch-tag-when-adding-a-git-submodule/18799234)). So I will just add the `master` branch. For future releases we need to create a `master-cp4` branch.

Reply:
:+1:

Reply:
Any reason for not doing `--recursive`  clone to save a step?

Reply:
I added this to the file: https://github.com/BodenmillerGroup/ImcSegmentationPipeline/blob/readimc_NE/README.md I guess it doesn't show up here?